cluster/swarm/cluster.go

```diff
@@ -119,7 +119,9 @@ func (c *Cluster) UnregisterEventHandler(h cluster.EventHandler) {
 func (c *Cluster) generateUniqueID() string {
 	for {
 		id := stringid.GenerateRandomID()
-		if c.Container(id) == nil {
+		if _, err := c.Container(id); err != nil {
+			continue
+		} else {
```

Comment:
This is not right. With this change, `err` is not equivalent to ID exist or not.